idmapping-rest/src/test/java/org/uniprot/api/idmapping/controller/IdMappingGroupByControllerIT.java

```diff
@@ -0,0 +1,174 @@
+package org.uniprot.api.idmapping.controller;
+
+import static org.hamcrest.Matchers.containsStringIgnoringCase;
+import static org.hamcrest.core.Is.is;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping;
+import org.uniprot.api.idmapping.common.JobOperation;
+import org.uniprot.api.idmapping.common.model.IdMappingJob;
+import org.uniprot.api.rest.download.model.JobStatus;
+import org.uniprot.store.indexer.DataStoreManager;
+import org.uniprot.store.search.document.Document;
+
+public abstract class IdMappingGroupByControllerIT {
+    protected static final String FREE_FORM_QUERY = "free-form-query";
+
+    @Value("${mapping.max.to.ids.with.facets.count}")
+    protected Integer maxToIdsWithFacetsAllowed;
+
+    @Autowired protected JobOperation idMappingResultJobOp;
+    @Autowired protected RequestMappingHandlerMapping requestMappingHandlerMapping;
+    private static final String INVALID_ORGANISM_ID = "36";
+    protected static final String CHEBI_ID = "12345678";
+
+    @Test
+    void getGroupBy_emptyResults() throws Exception {
```

Comment:
When passing some random parentId, APi is return resource not found.
e.g. https://wwwdev.ebi.ac.uk/uniprot/api/idmapping/wPFS7FOwiD/groups/taxonomy?parent=1
{
"url": "http://wwwdev.ebi.ac.uk/uniprot/api/idmapping/wPFS7FOwiD/groups/taxonomy",
"messages": [
"Resource not found"
]
}
It should return empty response